runelite-client/src/main/java/net/runelite/client/plugins/stealingartefacts/StealingArtefactsPlugin.java

```diff
@@ -0,0 +1,182 @@
+
+import com.google.common.collect.ImmutableSet;
+import lombok.AccessLevel;
+import lombok.Getter;
+import net.runelite.api.*;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.ItemContainerChanged;
+import net.runelite.api.events.NpcSpawned;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.callback.ClientThread;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+@PluginDescriptor(
+	name = "Stealing Artefacts",
+	description = "Show the location and status of your Stealing Artefact task.",
+	tags = {"thieving", "stealing", "artefact", "artefacts", "captain", "khaled"},
+	enabledByDefault = false
+)
+public class StealingArtefactsPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private StealingArtefactsOverlay overlay;
+
+	@Getter(AccessLevel.PACKAGE)
+	private StealingArtefactState stealingArtefactState;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean isInPortPiscariliusRegion;
+
+	private static final ImmutableSet<Integer> PORT_PISCARILIUS_REGIONS = ImmutableSet.of(6970, 7226);
+
+	private static final ImmutableSet<Integer> ARTEFACT_ITEM_IDS = ImmutableSet.of(ItemID.STOLEN_PENDANT,
+		ItemID.STOLEN_GARNET_RING, ItemID.STOLEN_CIRCLET, ItemID.STOLEN_FAMILY_HEIRLOOM, ItemID.STOLEN_JEWELRY_BOX);
+
+	private NPC captainKhaled = null;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlayManager.add(overlay);
+
+		if (client.getGameState() == GameState.LOGGED_IN)
+		{
+			clientThread.invoke(() ->
+				stealingArtefactState = StealingArtefactState.getStealingArtefactState(client.getVar(Varbits.STEALING_ARTEFACT_STATE)));
+		}
+		else
+		{
+			stealingArtefactState = null;
+		}
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		client.clearHintArrow();
+		overlayManager.remove(overlay);
+	}
+
+	private boolean hasArtefact()
+	{
+		ItemContainer inventory = client.getItemContainer(InventoryID.INVENTORY);
+
+		if (inventory != null)
+		{
+			final Stream<Item> items = Arrays.stream(inventory.getItems());
+			return items.anyMatch(item -> ARTEFACT_ITEM_IDS.contains(item.getId()));
+		}
+		return false;
+	}
+
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		int stealingArtefactStateVarbit = client.getVar(Varbits.STEALING_ARTEFACT_STATE);
+		StealingArtefactState state = StealingArtefactState.getStealingArtefactState(stealingArtefactStateVarbit);
+
+		if (state != StealingArtefactState.DELIVER_ARTEFACT)
+		{
+			stealingArtefactState = state;
+			if (state != StealingArtefactState.NO_TASK && state != StealingArtefactState.FAILURE)
```

Comment:
Compound if statements are easier to read if they're not negated. Swap the order of the blocks here:

```java
if (state == NO_TASK || state == FAILURE)
{
	client.clearHintArrow();
}
else
{
	client.setHintArrow(state.getWorldPoint());
}
```